src/analyze.ts

```diff
@@ -423,7 +423,15 @@ export async function runFinalize(
     delete process.env[sharedEnv.ODASA_TRACER_CONFIGURATION];
   }
 
-  fs.mkdirSync(outputDir, { recursive: true });
+  // After switching to Node16, this entire block can be replaced with `await fs.promises.rm(outputDir, { recursive: true, force: true });`.
+  try {
+    await fs.promises.rmdir(outputDir, { recursive: true });
```

Comment:
On windows, this might be finicky. I usually add retries and retry delay option when recursively deleting directories on windows.

https://nodejs.org/dist/latest-v12.x/docs/api/fs.html#fs_fs_rmdir_path_options_callback

```suggestion
    await fs.promises.rmdir(outputDir, { recursive: true, maxRetries: 5, retryDelay: 2000 });
```

The values above are pretty arbitrary, but might be good enough to avoid random errors.

Reply:
Makes sense, since Windows file locking can be pretty annoying. Interestingly `retryDelay` seems to only have been added in NodeJS v12.16.0, so I don't know how safe it is to depend on.

Reply:
Good point. I just verified locally, and it looks like node v10 will silently ignore `retryDelay` if it's passed.

Reply:
Hmm I can add it, but it's not recognized by TypeScript's typing declarations.